### PR TITLE
Reflection-free codegen for populating OpenGL functions

### PR DESCRIPTION
**Pull request by kekekeks**
This PR adds yet another roslyn generator.

NOT TESTED with Windows and macOS.

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022431-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]
Works as expected on Windows (with and without WGL), WASM and Android.

File: src/tools/DevGenerators/GetProcAddressInitialization.cs [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        /*
+        Console.WriteLine("PID: " + Process.GetCurrentProcess().Id);
+
+        File.WriteAllText("/tmp/pid", "PID: " + Process.GetCurrentProcess().Id);
+        while (!Debugger.IsAttached)
+        {
+            Thread.Sleep(1000);
+        }*/
```
Comment (maxkatz6) [inline src/tools/DevGenerators/GetProcAddressInitialization.cs:29]:
```suggestion
```

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022450-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by hez2010** [comment]
Wondering can this propogate the `[SuppressGCTransition]` attribute to function pointers (`delegate* unmanaged[CallConv, SuppressGCTransition]<T>`)? So we can avoid the GC context switch when we make frequent native calls that can return in a very short time.

**Comment by kekekeks** [comment]
From https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.suppressgctransitionattribute?view=net-7.0:

>The P/Invoke method that this attribute is applied to must have all of the following properties:
>
>-    Native function always executes for a trivial amount of time (less than 1 microsecond).
>-    Native function does not perform a blocking syscall (for example, any type of I/O).
>-    Native function does not call back into the runtime (for example, Reverse P/Invoke).
>-    Native function does not throw exceptions.
>-    Native function does not manipulate locks or other concurrency primitives.

We can't guarantee _any_ of these for OpenGL calls

**Pull request merged**